FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.HealthCheck/Checkers/AzureBlobStorageHealthChecker.cs

```diff
@@ -0,0 +1,85 @@
+    {
+        public const string HealthCheckBlobPrefix = ".healthcheck";
+        private readonly IAzureBlobContainerClient _blobContainerClient;
+
+        public AzureBlobStorageHealthChecker(
+            IAzureBlobContainerClientFactory azureBlobContainerClientFactory,
+            IOptions<JobConfiguration> jobConfiguration,
+            IOptions<DataLakeStoreConfiguration> storeConfiguration,
+            ILogger<AzureBlobStorageHealthChecker> logger)
+            : base(HealthCheckTypes.AzureBlobStorageCanReadWrite, false, logger)
+        {
+            EnsureArg.IsNotNull(azureBlobContainerClientFactory, nameof(azureBlobContainerClientFactory));
+            EnsureArg.IsNotNull(jobConfiguration, nameof(jobConfiguration));
+            EnsureArg.IsNotNull(storeConfiguration, nameof(storeConfiguration));
+
+            _blobContainerClient = azureBlobContainerClientFactory.Create(storeConfiguration.Value.StorageUrl, jobConfiguration.Value.ContainerName);
+        }
+
+        protected override async Task<HealthCheckResult> PerformHealthCheckImplAsync(CancellationToken cancellationToken)
+        {
+            var healthCheckResult = new HealthCheckResult(HealthCheckTypes.AzureBlobStorageCanReadWrite, false);
+            var blobPath = $"{HealthCheckBlobPrefix}";
+
+            try
+            {
+                // Ensure we can write to the storage account
+                byte[] bytes = Encoding.UTF8.GetBytes(HealthCheckBlobPrefix);
+                using MemoryStream stream = new (bytes);
+                await _blobContainerClient.UpdateBlobAsync(blobPath, stream, cancellationToken);
+            }
+            catch (Exception e)
+            {
+                healthCheckResult.Status = HealthCheckStatus.UNHEALTHY;
+                healthCheckResult.ErrorMessage = "Write content to blob failed." + e.Message;
+                return healthCheckResult;
+            }
+
+            try
+            {
+                // Ensure we can read from the storage account
+                var healthCheckStream = await _blobContainerClient.GetBlobAsync(blobPath, cancellationToken: cancellationToken);
+                string healthCheckContent;
+                healthCheckStream.Position = 0;
+                using StreamReader reader = new (healthCheckStream, Encoding.UTF8);
+                healthCheckContent = reader.ReadToEnd();
+
+                if (!Equals(healthCheckContent, HealthCheckBlobPrefix))
+                {
+                    healthCheckResult.Status = HealthCheckStatus.UNHEALTHY;
+                    healthCheckResult.ErrorMessage = "Read/Write content from blob failed.";
+                    return healthCheckResult;
+                }
+            }
+            catch (Exception e)
+            {
+                healthCheckResult.Status = HealthCheckStatus.UNHEALTHY;
+                healthCheckResult.ErrorMessage = "Read content from blob failed." + e.Message;
```

Comment:
Maybe we can use [string interpolation](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/tokens/interpolated) for error message value?